src/debugging/DockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,107 @@
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly dockerClient: DockerClient,
+        private readonly helpers: { [key in DockerPlatform]: DebugHelper }
+    ) { }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return undefined;
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration | undefined> {
+        const debugPlatform = getPlatform(debugConfiguration);
+        return callWithTelemetryAndErrorHandling(
+            `docker-launch/${debugPlatform || 'unknown'}`,
+            async (actionContext: IActionContext) => await this.resolveDebugConfigurationInternal(
+                {
+                    folder: folder || await quickPickWorkspaceFolder('To debug with Docker you must first open a folder or workspace in VS Code.'),
+                    platform: debugPlatform,
+                    actionContext: actionContext,
+                    cancellationToken: token,
+                },
+                debugConfiguration));
+    }
+
+    private async resolveDebugConfigurationInternal(context: DockerDebugContext, originalConfiguration: DockerDebugConfiguration): Promise<DockerDebugConfiguration | undefined> {
+        context.actionContext.telemetry.properties.platform = context.platform;
+
+        const helper = this.getHelper(context.platform);
+        const resolvedConfiguration = await helper.resolveDebugConfiguration(context, originalConfiguration);
+
+        if (resolvedConfiguration) {
+            await this.validateResolvedConfiguration(resolvedConfiguration);
+            await this.registerRemoveContainerAfterDebugging(resolvedConfiguration);
+        }
+
+        // TODO: addDockerSettingsToEnv?
+        return resolvedConfiguration;
+    }
+
+    private async validateResolvedConfiguration(resolvedConfiguration: ResolvedDebugConfiguration): Promise<void> {
+        if (!resolvedConfiguration.type) {
+            throw new Error('No debug type was resolved.');
+        } else if (!resolvedConfiguration.request) {
+            throw new Error('No debug request was resolved.');
+        }
+    }
+
+    private async registerRemoveContainerAfterDebugging(resolvedConfiguration: ResolvedDebugConfiguration): Promise<void> {
+        if (resolvedConfiguration.dockerOptions !== undefined
+            && (resolvedConfiguration.dockerOptions.removeContainerAfterDebug === undefined || resolvedConfiguration.dockerOptions.removeContainerAfterDebug)
+            && resolvedConfiguration.dockerOptions.containerNameToKill !== undefined) {
+            const runningContainerName =
+                (await this.dockerClient.listContainers({ format: '{{.Names}}' }))
+                    .split('\n')

```

Comment:
>split [](start = 21, length = 5)

Does this work on windows or do the strings end up with a trailing \r?

Reply:
I added some `.trim()`'s to the comparison to make sure any trailing/leading whitespace gets removed.